parse.py

```diff
@@ -513,10 +513,13 @@ def int_convert(base=None):
     it overrides the default base of 10.
 
     It may also have other non-numeric characters that we can ignore.
-    '''
-    CHARS = '0123456789abcdefghijklmnopqrstuvwxyz'
+    '''    
+
+    def __init__(self, base=None):
+        self.base = base
+        self.CHARS = '0123456789abcdefghijklmnopqrstuvwxyz'
```

Comment:
CHARS doesn't need to move to be an instance attribute - leaving it where it was will make it a class attribute.